liwords-ui/src/shared/follower.tsx

```diff
@@ -0,0 +1,74 @@
+import {
+  useFriendsStoreContext,
+  useLoginStateStoreContext,
+} from '../store/store';
+import axios from 'axios';
+import { toAPIUrl } from '../api/api';
+import React from 'react';
+import { notification } from 'antd';
+
+type FollowerProps = {
+  className?: string;
+  target: string;
+  tagName?: string;
+  friendCallback?: () => void;
+};
+
+export const TheFollower = (props: FollowerProps) => {
+  const { friends, setPendingFriendsRefresh } = useFriendsStoreContext();
+  const { loginState } = useLoginStateStoreContext();
+  const { userID } = loginState;
+
+  if (userID === props.target) {
+    return null;
+  }
+
+  let apiFunc: string;
+  let friendText: string;
+  if (friends[props.target]) {
+    apiFunc = 'Remove';
+    friendText = 'Remove from friends';
+  } else {
+    apiFunc = 'Add';
+    friendText = 'Add friend';
+    // Add some confirmation.
+  }
+
+  const friendAction = () => {
+    axios
+      .post(
+        toAPIUrl('user_service.SocializeService', `${apiFunc}Follow`),
```

Comment:
apiFunc should be the full text (AddFollow/RemoveFollow) so it's easier to grep for AddFollow next time

Reply:
Keeping it consistent with Blocker (Cesar's code) for now, but you can submit a new pr changing both.